exonum/src/api/mod.rs

```diff
@@ -198,34 +198,16 @@ pub trait Api {
         Ok(value)
     }
 
-    /// Deserializes an optional parameter from request body or get-parameters
-    fn optional_param<T>(&self, request: &mut Request, name: &str) -> Result<Option<T>, ApiError>
+    /// Deserializes request's body as a struct of type `T`.
+    fn parse<T: 'static>(&self, req: &mut Request) -> Result<T, ApiError>
```

Comment:
Nitpick: `parse` is a rather non-descriptive name; I'd suggest something like `parse_body`.